src/core/manager/mod.rs

```diff
@@ -1681,6 +1681,54 @@ mod tests {
     layout_trigger_test!(position_client; true; 10, Region::default(), true);
     layout_trigger_test!(layout_screen; true; 0);
 
+    #[test]
+    fn layout_trigger_test_follow_focus_cycle_client() {
+        let conn = RecordingXConn::init();
+        let conf = Config {
+            layouts: focus_test_layouts(true),
+            ..Default::default()
+        };
+        let mut wm = WindowManager::new(conf, conn, vec![], logging_error_handler());
+        wm.init().unwrap();
+        add_n_clients(&mut wm, 3, 0);
+        wm.focus_workspace(&Selector::Index(1)).unwrap();
+        add_n_clients(&mut wm, 3, 30);
+        wm.focus_workspace(&Selector::Index(0)).unwrap();
+        wm.conn.clear();
+        wm.cycle_client(Direction::Forward).unwrap();
+
+        let should_be_true = wm.conn.calls().iter().any(|c| c.0 == *"position_client");
+
+        assert!(should_be_true);
+    }
+
+    #[test]
+    fn layout_trigger_test_follow_focus_focus_client() {
+        let conn = RecordingXConn::init();
+        let conf = Config {
+            layouts: focus_test_layouts(true),
+            ..Default::default()
+        };
+        let mut wm = WindowManager::new(conf, conn, vec![], logging_error_handler());
+        wm.init().unwrap();
+        wm.handle_map_request(0).unwrap();
+        wm.handle_map_request(1).unwrap();
+        wm.focus_client(&Selector::WinId(0)).unwrap();
+        wm.conn.clear();
+
+        // shouldn't trigger layout when re-focusing same client
+        wm.focus_client(&Selector::WinId(0)).unwrap();
+        let should_be_false = wm.conn.calls().iter().any(|c| c.0 == *"position_client");
+        wm.conn.clear();
+
+        // focusing any other client should trigger layout stuff
+        wm.focus_client(&Selector::WinId(1)).unwrap();
+        let should_be_true = wm.conn.calls().iter().any(|c| c.0 == *"position_client");
+
+        assert!(!should_be_false);
```

Comment:
The naming and negation is a bit confusing here. Can you inline these assertions please? (See the example given above)